src/compiler/scale.ts

```diff
@@ -67,7 +68,6 @@ export function type(fieldDef: FieldDef, channel: Channel, mark: Mark): string {
       if (fieldDef.scale.type !== undefined) {
         return fieldDef.scale.type;
       }
-      // TODO: add timeUnit for other timeUnit once added
       switch (fieldDef.timeUnit) {
```

Comment:
better put switch inside the `if(fieldDef.timeUnit)` below and `return 'ordinal';` for `default:` case (yearmonth, monthday, ...)